src/crafting.h

```diff
@@ -104,6 +104,84 @@ struct recipe {
 
 };
 
+/**
+*   enum used by comp_selection to indicate where a component should be consumed from.
+*/
+enum usage {
+    use_from_map = 1,
+    use_from_player = 2,
+    use_from_both = 1 | 2,
+    use_from_none = 4,
+    cancel = 5 // FIXME: hacky.
```

Comment:
So it's like 'control bits'. Where each bit is used as a control signal. In that case, I'm used to seeing something like this:
```c++
enum usage {
    use_from_map = ( 1U << 0 ),
    use_from_player = ( 1U << 1 ),
    use_from_both = use_from_map | use_from_player,
    use_from_none = ( 1U << 2 ),
    cancel = ( 1U << 3 )
};
```
It's still good to know, I changed the use of ```&``` to ```==``` because it kept failing with ```5```. For SOME reason I thought that ```1 & 5``` would be 0: ```101 & 001 -> 000```, it must have been late at night :P

Looking at it now, this will create a problem with consuming items from both locations. Where no items will be consumed at all. :$ oops.